File: lib/file.dart
import 'package:az_player_plugin/az_player_plugin.dart';

class File implements InterfaceFile{
  @override
  final num currentTime;

  @override
  final FileStatus fileStatus;

  @override
  final String fileURL;

  @override
  final String image;

  @override
  final int pk;

  @override
  final String title;

  File(this.pk, this.fileURL, this.title, this.currentTime, this.fileStatus, this.image);

  @override
  Map<String, dynamic> toJson() {
    
    Map<String, dynamic> map = Map();
    map['currentTime'] = currentTime;
    map['fileStatus'] = fileStatus.value;
    map['fileURL'] = fileURL;
    map['image'] = image;
    map['pk'] = pk;
    map['title'] = title;
    
    return map;
  }
	
}

File: android/src/main/kotlin/pro/zahedi/flutter/plugin/player/az_player_plugin/File.java
package pro.zahedi.flutter.plugin.player.az_player_plugin;

import java.util.Map;

public class File {

    int pk;
    String title;
    double currentTime;
    String fileURL;
    // image path
    String image;
    FileStatus fileStatus = FileStatus.ready;

    File(int pk, String title, double currentTime, String fileURL, String image, FileStatus fileStatus){
        this.pk = pk;
        this.title = title;
        this.currentTime = currentTime;
        this.fileURL = fileURL;
        this.image = image;
        this.fileStatus = fileStatus;
    }

    public static File fromJson(Map<String, Object> model)
    {
       FileStatus fileStatus = FileStatus.ready;
       return new File((int) model.get("pk"),
               (String) model.get("title"),
               Double.parseDouble(model.get("currentTime").toString()),
               (String) model.get("fileURL"),
               (String) model.get("image"),
               fileStatus.fromRawValue((Integer) model.get("fileStatus")));
    }
}
